overlord/devicestate/handlers_install.go

```diff
@@ -312,6 +313,15 @@ func checkEncryption(model *asserts.Model) (res bool, err error) {
 		return false, nil
 	}
 
+	// check if kernel has fde-setup hook based encryption support
+	if kernelInfo, err := snapstate.KernelInfo(st, deviceCtx); err == nil {
+		// XXX: should we run the fde-setup hook with
+		//      "op":"available" or similar now?
```

Comment:
maybe we could use op: features for this as well?

Reply:
Oh, I like this idea - so you are suggesting that if "op":"features" returns an error we would treat this as "fde-setup" will not work on this specific system?

Reply:
something like that, I'm happy to discuss possible details

Reply:
From the discussion with Samuele:
```
happy case: {"features": []} 
unsupported: {"error": "hardware-unsupported"}
```
if the hook returns an non-zero exit code we will also consider it unsupported and log the error.